static/app/components/forms/fieldFromConfig.tsx

```diff
@@ -32,74 +34,67 @@ type Props = {
-        return <EmailField {...props} />;
-      case 'hidden':
-        return <HiddenField {...props} />;
-      case 'string':
-      case 'text':
-      case 'url':
-        if (props.multiline) {
-          return <TextareaField {...props} />;
-        }
-        return <TextField {...props} />;
-      case 'number':
-        return <NumberField {...props} />;
-      case 'textarea':
-        return <TextareaField {...props} />;
-      case 'choice':
-      case 'select':
-      case 'array':
-        return <SelectField {...props} />;
-      case 'choice_mapper':
-        // TODO(ts) The switch on field.type is not resolving
-        // the Field union for this component. The union might be 'too big'.
-        return <ChoiceMapperField {...(props as any)} />;
-      case 'radio':
-        const choices = props.choices;
-        if (!Array.isArray(choices)) {
-          throw new Error('Invalid `choices` type. Use an array of options');
-        }
-        return <RadioField {...props} choices={choices} />;
-      case 'table':
-        // TODO(ts) The switch on field.type is not resolving
-        // the Field union for this component. The union might be 'too big'.
-        return <TableField {...(props as any)} />;
-      case 'project_mapper':
-        return <ProjectMapperField {...props} />;
-      case 'sentry_project_selector':
-        return <SentryProjectSelectorField {...props} />;
-      case 'select_async':
-        return <SelectAsyncField {...props} />;
-      case 'custom':
-        return field.Component(props);
-      default:
-        return null;
-    }
+  switch (field.type) {
+    case 'separator':
+      return <FieldSeparator />;
+    case 'secret':
+      return <InputField {...(componentProps as InputFieldProps)} type="password" />;
+    case 'range':
+      return <RangeField {...(componentProps as RangeFieldProps)} />;
+    case 'blank':
+      return <BlankField {...(componentProps as FieldProps)} />;
+    case 'bool':
+    case 'boolean':
+      return <BooleanField {...(componentProps as BooleanFieldProps)} />;
+    case 'email':
+      return <EmailField {...(componentProps as EmailFieldProps)} />;
+    case 'hidden':
+      return <HiddenField {...(componentProps as HiddenFieldProps)} />;
+    case 'string':
+    case 'text':
+    case 'url':
+      if (props.field.multiline) {
+        return <TextareaField {...(componentProps as TextareaFieldProps)} />;
+      }
+      return <TextField {...componentProps} />;
+    case 'number':
+      return <NumberField {...componentProps} />;
+    case 'textarea':
+      return <TextareaField {...(componentProps as TextareaFieldProps)} />;
+    case 'choice':
+    case 'select':
+    case 'array':
+      return <SelectField {...(componentProps as SelectFieldProps<any>)} />;
+    case 'choice_mapper':
+      return <ChoiceMapperField {...(componentProps as ChoiceMapperFieldProps)} />;
+    case 'radio':
+      if (Array.isArray(props.field.choices)) {
```

Comment:
was this wrong before?

Before:
`props.choices`
After:
`props.field.choices`

Reply:
Let me double check, I may have made a mistake

Reply:
Yup, it's actually componentProps.multiline and componentProps.choices here. Nice catch